client/rpc-api/src/state/mod.rs

```diff
@@ -263,6 +263,11 @@ pub trait StateApi<Hash> {
 	/// [2]: https://www.shawntabrizi.com/substrate/transparent-keys-in-substrate/
 	/// [3]: https://www.shawntabrizi.com/substrate/querying-substrate-storage-via-rpc/
 	///
+	/// - `methods` (param index 3): String of comma separated (no spaces) tracing event method.
+	/// If an empty string is specified no events will be filtered out. If anything other than
+	/// an empty string is specified, events will be filtered by method (so non-method events will
+	/// **not** show up).
```

Comment:
What happens if the caller specifies both `storage_keys` and `methods`? Is there a priority? Or `Invalid Params` error (this would be my preference)? Either way it'd be good to document the behaviour.

Reply:
Dumb question: what is the expected format of these methods? Is it the Rust name or some other convention (JS?)? A short example here would be valuable to newcomers I think.

Reply:
> What happens if the caller specifies both `storage_keys` and `methods`? Is there a priority? 

I prefer the priority of filter conditions to be determined according to the order of declaration of rpc parameters. (Currently, `storage_keys` is filtered first, then methods)

> what is the expected format of these methods? 

These `methods` are just ordinary string format, you could find these methods by `rg "method = " primitives/state-machine/src/ext.rs`

> A short example here would be valuable to newcomers I think.

Yes, I will add a simple example into doc

Reply:
> I prefer the priority of filter conditions to be determined according to the order of declaration of rpc parameters. (Currently, `storage_keys` is filtered first, then methods)

Fair. Can you document that behaviour please?